### ISSUE TITLE
Allocate like (new Dataset with one or more different parameters)

### ISSUE THREAD

**Issue by mohit28290** [zowe/vscode-extension-for-zowe#1960]
**Is your feature request related to a problem? Please describe.**

Allocate like feature enhancement, when allocating new dataset like existing one with one or more different parameters example new volume name.

**Describe the solution you'd like**

Allocate like feature with another option to specify or alter few existing attributes.

**Describe alternatives you've considered**

When allocate like selected for a dataset it only ask for new DS name, if tab could show the attributes in edit mode allowing alteration of few parameters.

**Additional context**

![image](https://user-images.githubusercontent.com/114663645/195116461-d47acbf1-937d-4ba0-a586-36b7520eb6a3.png)

**Comment by MikeBauerCA** [comment]
If this is prioritized, I would recommend looking at how the Zowe Explorer for IntelliJ implemented "Allocate Like". When you do "allocate like", it opens up a form like this:
![image](https://user-images.githubusercontent.com/35504792/214309395-6995db62-9a2c-4002-a479-82feb5982101.png)

Another aspect of the plug-in that I like is that it shows the volume next to the data set names in the list.

I used the allocate-like function of Zowe Explorer and it placed my dataset on a wrk volume which was deleted after a few days.  I do get this same behavior when running the "allocate dataset(public.jcl2) like (public.jcl)" command on TSO. However, I wasn't aware of that and because ZE doesn't show the volume, I would have lost my work had I not caught it.

**Comment by RichardSGFR** [comment]
I agree with @MikeBauerCA, but I would add that in the "Advanced Parameters" the Storage Class should be prefilled with the Storage Class of the initial PDS. Which would lead to avoid this issue : https://github.com/zowe/zowe-explorer-vscode/issues/3332

**Comment by zFernand0** [comment]
Hey all,

After doing some investigation, it'd seem that we don't have a good way to retrieve the storage class information from the source/initial PDS in order to maintain that same storage class when allocating the target/new PDS.

After attempting a few different approaches, I believe it may be possible to guarantee the same volume by accessing the DFSMS utility directly (via TSO commands) after the target PDS is created.
